prog/weaveproxy/main.go

```diff
@@ -15,6 +15,23 @@ var (
 	defaultListenAddrs = []string{"tcp://0.0.0.0:12375", "unix:///var/run/weave.sock"}
 )
 
+type listOpts struct {
+	values *[]string
```

Comment:
Actually there's a worse problem here, which is that the listOpts is purely additive, so if you supply an argument it gets _added_ to the default rather than replacing it.